src/smolagents/agents.py

```diff
@@ -565,21 +578,21 @@ def planning_step(self, task, is_first_step: bool, step: int) -> None:
             # Log final facts and plan
             final_plan_redaction = textwrap.dedent(
                 f"""I still need to solve the task I was given:
```

Comment:
Here guys use `texwrap.dedent` so that we don't need to format heredoc strings this way, instead maintaining leading whitespaces on each line.

Reply:
Thank you, that makes sense; added `texwrap.dedent` everywhere.